src/core/lib/layer.js

```diff
@@ -72,7 +72,7 @@ const defaultProps = {
 let counter = 0;
 
 export default class Layer {
-  constructor(props) {
+  constructor(...props) {
```

Comment:
why do we need spread operator here?

Reply:
This allows us to accept a variable number of arguments and capture them as an `Array`.